api/src/database/migrations/20251027A-add-provider-access.ts

```diff
@@ -0,0 +1,46 @@
+import type { Knex } from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+	const rowsLimit = 100;
+	let rowsOffset = 0;
+	let hasMore = true;
+
+	while (hasMore) {
+		const policies = await knex('directus_permissions')
+			.select('directus_permissions.id', 'directus_permissions.fields')
+			.join('directus_policies', 'directus_policies.id', 'directus_permissions.policy')
+			.where('directus_policies.app_access', '=', 1)
+			.andWhere('directus_permissions.collection', '=', 'directus_users')
+			.andWhere('directus_permissions.action', '=', 'read')
+			.andWhereNot('directus_permissions.fields', 'LIKE', '*')
+			.limit(rowsLimit)
+			.offset(rowsOffset);
+
+		if (policies.length === 0) {
```

Comment:
We can further improve this by adding `|| policies.length < rowsLimit` and moving it under the for loop. This will reduce the loop by one ensuring we do query an extra time when we know we have reached the end of the line.